src/components/tx-flow/SafeTxProvider.tsx

```diff
@@ -42,11 +42,12 @@ const SafeTxProvider = ({ children }: { children: ReactNode }): ReactElement =>
   const isSigned = safeTx && safeTx.signatures.size > 0
 
   // Recommended nonce and safeTxGas
-  const recommendedNonce = Math.max(safe.nonce, useRecommendedNonce() ?? 0)
+  const recommendedNonce = useRecommendedNonce()
+  const safeRecommendedNonce = recommendedNonce ? Math.max(safe.nonce, recommendedNonce) : undefined
```

Comment:
IMHO this should be part of useRecommendedNonce

Reply:
True, I've moved it into the hook.